### PR TITLE
Type added: "char**" to SDB, ref #18633

### PR DESCRIPTION
**Pull request by Shadorain**
**Checklist**

- [X] Closing issue: #18633
- [X] Ready to merge
- [X] I've added tests

**Tests**

```bash
:> ts char **
pf pp

:> t
char
char *
char **
double
...
```

Removes `stderr` output in Visual Panel mode when scrolling past a `char **` (shown in issue)

**Description**

The purpose of this Pull Request is to address a missing standard C type in SDB that should be added and cleans up some issues I have personally with not having it and for others. I simply changed one file (three lines) and added the type so it isn't much of a contribution but it should help somewhat.

### CODE REVIEW

**Comment by Shadorain** [comment]
Let me know if I need to edit the PR in any way so that I can make it better for you. It is my first on this project.

**Comment by Shadorain** [comment]
How does this simple working change throw two errors in testing? 
I can't understand the errors so could someone help on why this is failing?

**Comment by trufae** [comment]
I think the valid type definition for char ** should be something like: `*z` instead of `p`.

See:
```
[0x00000000]> wv 32
[0x00000000]> w hello world @ 32
[0x00000000]> pf *z
(*0x20)0x00000000 = "hello world"
[0x00000000]>
```

**Comment by trufae** [comment]
About the broken tests. you should run the testsuite failing files with `r2r -i test/db/types`


Screenshot 2021-04-26 at 20 52 56

Can you add a test for pf *z and another test for the `char **` type definition usage? maybe we should have a type for `pf` to handle arrays of strings ended with NULL. But blindly char ** can be used for passing a reference to a char*, or also to define a pointer to a list of pointers to strings `pf *[4]z..` but this is not currently supported.

**Comment by Shadorain** [comment]
> About the broken tests. you should run the testsuite failing files with `r2r -i test/db/types`
> 
> Can you add a test for pf _z and another test for the `char **` type definition usage? maybe we should have a type for `pf` to handle arrays of strings ended with NULL. But blindly char ** can be used for passing a reference to a char_, or also to define a pointer to a list of pointers to strings `pf *[4]z..` but this is not currently supported.

This is all updated with `*z` instead of p now
Okay cool, didn't know that suite I had locally, so I tried that `r2r -i test/db/cmd/types` and...:
```
#####################
 36 failed test(s) 🚨
#####################
```
So that isn't good, most the errors look to be `Cannot open 'bins/pe/pe.exe'` etc, that sort of problem though. 

By add a test I don't fully know what you mean so I'll give the output of when I run that (with my commit):

```bash
[0x00000750]> pf z
0x00000750 = "1\xedI\x89\xd1^H\x89\xe2H\x83\xe4\xf0PTL\x8d\x05\n\x04"
[0x00000750]> pf *z
(*0x89485ed18949ed31)0x00000750 = "\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff"
[0x00000750]> pf *[4]z..
(*0x89485ed18949ed31)0x00000758 = "\xe2H\x83\xe4\xf0PTL\x8d\x05\n\x04"
[0x00000750]> wv 64   # btw idk why this doesn't work here either...
Failed to write
```

**Comment by trufae** [comment]
You didnt fetched the test/bins repository . Try with “make tests”. In the github ci logs search for XX]

**Comment by trufae** [comment]
I'll make r2r clone the bins repo if the directory is not found , that will save some time for new users :)

**Comment by Shadorain** [comment]
> I'll make r2r clone the bins repo if the directory is not found , that will save some time for new users :)

yes i believe that would be beneficial

> You didnt fetched the test/bins repository . Try with “make tests”. In the github ci logs search for XX]

Okay so I tried this and ran `make tests` and this errored but it looks like it made enough because now here are the only three failed tests when testing `r2r -i test/db/cmd/types`:
image

**Comment by trufae** [comment]
Yes, that purput looks correct. After showing the failing tests the -i flag should make it interactive and prompt you to fix the tests so you can commit them later. Dont worry for late responses :) thanks!

**Review by lastthyme — APPROVED** [review]

**Review by lastthyme — APPROVED** [review]

**Review by lastthyme — APPROVED** [review]

**Comment by trufae** [comment]
There are only 6 broken tests, just run r2r -i and press 'f' to fix the tests when prompted and update the pr

**Comment by Shadorain** [comment]
> There are only 6 broken tests, just run r2r -i and press 'f' to fix the tests when prompted and update the pr

Yep I'm actually doing this right now, its all fixed and I am pushing up in a sec

**Comment by trufae** [comment]
uhmgf seems like the pr got messed up with several unrelated reverted changes

**Comment by trufae** [comment]
Dont mark the pr as draft because the ci wont run otherwise. But you should push in this branch only the commits of your fix not other unrelated reverts. As a recommendation never use merge commits in prs

**Comment by trufae** [comment]
Also, who is @lastthyme and why aproves this??

**Comment by trufae** [comment]
> > There are only 6 broken tests, just run r2r -i and press 'f' to fix the tests when prompted and update the pr
> 
> Yep I'm actually doing this right now, its all fixed and I am pushing up in a sec
> The `libr/` generated files I added to the commit, not sure if I am supposed to (these are made when installing `./sys/install.sh`) but I can remove them if necessary.

Uhm so maybe sys/install caused the git mess?

**Comment by Shadorain** [comment]
> Dont mark the pr as draft because the ci wont run otherwise. But you should push in this branch only the commits of your fix not other unrelated reverts. As a recommendation never use merge commits in prs

Gotcha, I wont do that again, I can revert if you'd like?

> Also, who is @lastthyme and why aproves this??

I have absolutely no idea, I was so confused when I got the notifications of this

> > > There are only 6 broken tests, just run r2r -i and press 'f' to fix the tests when prompted and update the pr
> > 
> > 
> > Yep I'm actually doing this right now, its all fixed and I am pushing up in a sec
> > The `libr/` generated files I added to the commit, not sure if I am supposed to (these are made when installing `./sys/install.sh`) but I can remove them if necessary.
> 
> Uhm so maybe sys/install caused the git mess?

could be I can go through and revert. fine by me if that would be helpful

**Comment by Shadorain** [comment]
> Dont mark the pr as draft because the ci wont run otherwise. But you should push in this branch only the commits of your fix not other unrelated reverts. As a recommendation never use merge commits in prs

yeah for sure, I won't again. This is my first PR to something useful so I'm learning the ropes, thanks for being patient with me.

> Also, who is @lastthyme and why aproves this??

I have absolutely no idea, i was really confused when getting the notifications from that.

> > > There are only 6 broken tests, just run r2r -i and press 'f' to fix the tests when prompted and update the pr
> > 
> > 
> > Yep I'm actually doing this right now, its all fixed and I am pushing up in a sec
> > The `libr/` generated files I added to the commit, not sure if I am supposed to (these are made when installing `./sys/install.sh`) but I can remove them if necessary.
> 
> Uhm so maybe sys/install caused the git mess?

This could be the case, i am going to revert and see what I can do to fix it

**Comment by Shadorain** [comment]
Good to see there are no more errors in the tests. Thoughts or things I can change before merging?

File: libr/core/.vim-bookmarks [hunk 1]
```diff
@@ -0,0 +1,3 @@
+let l:bm_file_version = 1
```
Comment (trufae) [inline libr/core/.vim-bookmarks:1]: dont commit this file

File: test/db/cmd/cmd_k [hunk 2]
```diff
@@ -99,6 +99,7 @@ EXPECT=<<EOF
       "ceill=func",
```
Comment (trufae) [inline test/db/cmd/cmd_k:99]: dont commit this test/.null.bin either

**Review by trufae — CHANGES_REQUESTED** [review]

**Comment by Shadorain** [comment]
Had no idea those were still there. Thought I removed those, thanks Ill fix that.

**Comment by trufae** [comment]
Congrats! your first contribution is now merged! :) thanks for the patience fixing all those little things

**Comment by Shadorain** [comment]
You're the best, thank you for your patience too, didnt e mean to be such a problem lol

**Pull request merged**